### PR TITLE
Added checks for the output sizes in FractionalMaxPool2d and FractionalMaxPool3d

### PR DESCRIPTION
**Pull request by izdeby**
Fixing #18705
The problem was with output width parameter which didnt have a proper check and this can lead to a situation when we have division by 0. 

Fix:
- Added output width check

Test: 
- Added test case

### CODE REVIEW

**Review by ssnl — COMMENTED** [review]
test? :)

**Review by ezyang — APPROVED** [review]
Tests please!

**Comment by izdeby** [comment]
@gchanan @ezyang, i couldn't repro the bug. The only way to get to that code path where have division by 0 is if scalar type is float and it gives "-nan" but not an error. As a workaround, we decided to avoid the division completely. Its possible for CPU part but not that straightforward for the CUDA part. 

For CPU: see changes in this PR
For CUDA: see [this](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/cuda/FractionalMaxPool3d.cu#L31-L38). 

What would you suggest doing for CUDA?

**Comment by ezyang** [comment]
While the function you've linked is called on device (in which case, it is indeed hard to propagate errors), the division happens entirely on values which are known on host (outputSize). So I would just add an extra check in host code about output size and then rely on that invariant being established to avoid division by zero on cuda. Does that make sense?

File: aten/src/ATen/native/FractionalMaxPool2d.cpp [hunk 1]
```diff
@@ -166,6 +166,10 @@ void fractional_max_pool2d_out_cpu_template(
   TORCH_CHECK(outputW + poolSizeW - 1 <= inputW,
     "fractional_max_pool2d(): pool width ", poolSizeW,
     " too large relative to input width ", inputW);
+  TORCH_CHECK(outputW > 1,
```
Comment (SsnL) [inline aten/src/ATen/native/FractionalMaxPool2d.cpp]: I think one of these should check height?

File: aten/src/ATen/native/FractionalMaxPool3d.cpp [hunk 2]
```diff
@@ -188,6 +188,10 @@ void fractional_max_pool3d_out_cpu_template(
   TORCH_CHECK(outputH + poolSizeH - 1 < inputH,
            "fractional_max_pool3d_out(): pool height ", poolSizeH,
            " too large relative to input height ", inputH);
+  TORCH_CHECK(outputW > 1,
+           "fractional_max_pool3d_out(): output width has to be longer than 1");
+  TORCH_CHECK(outputW > 1,
```
Comment (SsnL) [inline aten/src/ATen/native/FractionalMaxPool3d.cpp]: Should there be three checks for H, W, and D?

File: test/test_nn.py [hunk 3]
```diff
@@ -3155,6 +3155,9 @@ def func(x):
         gradcheck(func, [x])
         gradgradcheck(func, [x])
 
+        x = torch.randn(2, 2, 2)
+        self.assertRaises(RuntimeError, lambda: F.fractional_max_pool2d(x, (1, 1), output_size=(3, 3)))
```
Comment (SsnL) [inline test/test_nn.py]: Hmmm, does this actually successfully trigger the error message? You have `output_size = (3, 3)` here, which are `> 1`.

**Pull request closed without merging**